sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java

```diff
@@ -137,8 +137,8 @@ void waitForDone() throws Exception {
               throw e;
             }
           } else {
-            // Job failed, schedule it again.
-            LOG.info("Job {} failed. retrying.", jobInfo.pendingJob.currentJobId);
+            // Job not yet complete, schedule it again.
```

Comment:
Hmm, seems like pollJob() call usually returns false if the job failed.
https://github.com/apache/beam/blob/2a4092dfb8c46408818402b1c4a09a8cd44e907a/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java#L260

Reply:
According to actual job logs, the most common reason is that the job hasn't started yet, which is this line:
https://github.com/apache/beam/blob/2a4092dfb8c46408818402b1c4a09a8cd44e907a/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java#L268